CAPTCHA - Clarify content in OIE overview (#4496)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/concepts/oie-intro/index.md

<ApiLifecycle access="ie" />

Okta Identity Engine is a new authentication pipeline that provides valuable new features and a more flexible approach to your auth needs. This article provides a high-level introduction.

This page discusses the following:

* New features Identity Engine brings to the table
* The deployment models that use these features
* Changes to the documentation experience to support Identity Engine

> **Note**: If you’re an admin, or are looking for product docs related to Identity Engine, see the Identity Engine [Get started page](https://help.okta.com/okta_help.htm?type=oie&id=ext-get-started-oie).

## Identity Engine new features

Identity Engine unlocks many new capabilities.

### App context in email templates

> Identity Engine makes the app context available when a user enters an authentication flow. Find context variables in our email templates. The variables allow customers to dynamically customize email style and content based on the app that triggers an email notification.

Reviewer: ```suggestion
Identity Engine makes the app context available when a user enters an authentication flow. Find context variables in our email templates. These variables allow customers to dynamically customize email style and content based on the app that triggers an email notification.
```

### CAPTCHA

CAPTCHA is a well-known strategy for mitigating attacks by bots. Identity Engine offers integrations with market-leading CAPTCHA services for registration, sign-in, and account recovery.

> Okta only supports the following CAPTCHA services. Using any other type could lead to lockout:

Author: @lesterchoi-okta I've made some updates. What do you think?

> Okta only supports the following CAPTCHA services. Using any other type could lead to lockout:

Reviewer 2: Do you need/want to mention that they'll need to contact support to restore access in case of lockout?

> Okta only supports the following CAPTCHA services. Using any other type could lead to lockout:

Author: Good point. Def worth mentioning.

* **Embedded Sign-In Widget**: Embed the Sign-In Widget into your own code base to handle the authentication on your servers. This provides a balance between complexity and customization.
* **Embedded SDK-driven sign-in flow**: Use our SDKs to create a custom authentication experience. This option is the most complex and leaves you with the most responsibility, but offers the most control.

See [Okta deployment models &mdash; redirect vs. embedded](/docs/concepts/redirect-vs-embedded/) for an overview of the different deployment models, and see [Sign users in](/docs/guides/sign-in-overview/) for implementation details.

## SDKs and sample apps

Okta has a host of SDKs available for integrating new Identity Engine features into your apps using [Okta deployment models](#authentication-deployment-models), and sample apps to show them in action.

* [Browse our SDKs and samples](/code/)
* [Set up and explore our Identity Engine sample apps](/docs/guides/oie-embedded-common-download-setup-app/)

## Identity Engine versus Classic Engine documentation approach

> In our documentation, Okta are moving towards supporting Identity Engine by default, while still providing information for Classic Engine users.

Reviewer: ```suggestion
In our documentation, Okta is moving towards supporting Identity Engine by default, while still providing information for Classic Engine users.
```

* Pages and page sections covering features that only work in Identity Engine have a blue Identity Engine banner at the top.
* Content that works in both Identity Engine and Classic Engine have no banner. Any slight differences are covered in the page text.
* Content written for Classic Engine that won't work in Identity Engine has a note at the top that explains what the issue is, and, if appropriate, where Identity Engine users can go to find support.

> * For guides that were extensively updated to support Identity Engine, Okta keeps a [Classic Engine](/docs/guides/archive-overview/) version so it's still accessible if needed.

Reviewer: ```suggestion
* For guides that were extensively updated to support Identity Engine, Okta keeps a [Classic Engine](/docs/guides/archive-overview/) version available if needed.
```
